src/main/java/uk/gov/hmcts/reform/sendletter/entity/LetterRepository.java

```diff
@@ -14,6 +15,8 @@
     @Lock(LockModeType.PESSIMISTIC_WRITE)
     Stream<Letter> findByState(LetterState state);
 
+    Stream<Letter> findByStateAndSentToPrintAtBeforeAndIsFailedFalse(LetterState state, Timestamp before);
```

Comment:
no need to read entire pdf from Db, you can use projections: https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#projections